lib/src/source/hosted.dart

```diff
@@ -1100,3 +1114,95 @@ class _RefAndCache {
   @override
   bool operator ==(Object other) => other is _RefAndCache && other.ref == ref;
 }
+
+extension GCSChecksumValidation on http.StreamedResponse {
```

Comment:
Do we need an extension? Can't we do with utility methods..

Reply:
I made it an extension primarily because of how it grouped all the checksum related logic, and I had assumed that it would make extensibility (checksum validation for other hosts) and testing easier.

I can break it up into utility methods, though. I also see now that checksum validation seems like it will be more of a first-class feature for all hosts in the future, so the extension makes less sense.